src/metatensor/models/cli/train.py

```diff
@@ -288,27 +317,46 @@ def _train_model_hydra(options: DictConfig) -> None:
         if options["seed"] is not None:
             generator.manual_seed(options["seed"])
 
-        train_dataset, validation_dataset = _train_test_random_split(
-            train_dataset=train_dataset,
-            train_size=train_size,
-            test_size=validation_size,
-            generator=generator,
-        )
+        for i_dataset, train_dataset in enumerate(train_datasets):
+            train_dataset_new, validation_dataset = _train_test_random_split(
+                train_dataset=train_dataset,
+                train_size=train_size,
+                test_size=validation_size,
+                generator=generator,
+            )
+
+            train_datasets[i_dataset] = train_dataset_new
+            validation_datasets.append(validation_dataset)
     else:
-        validation_options = expand_dataset_config(validation_options)
-        validation_structures = read_structures(
-            filename=validation_options["structures"]["read_from"],
-            fileformat=validation_options["structures"]["file_format"],
-            dtype=torch.get_default_dtype(),
-        )
-        validation_targets = read_targets(
-            conf=validation_options["targets"], dtype=torch.get_default_dtype()
-        )
-        validation_dataset = Dataset(
-            structure=validation_structures,
-            energy=validation_targets["energy"],
+        validation_options_list = expand_dataset_config(validation_options)
+        check_options_list(validation_options_list)
+
+        if len(validation_options_list) != len(train_options_list):
+            raise ValueError(
+                f"Validation dataset with length {len(validation_options_list)} has "
+                "a different size than the train datatset with length "
+                f"{len(train_options_list)}."
+            )
+
+        check_units(
+            actual_options=validation_options_list, desired_options=train_options_list
```

Comment:
I don't think this check is strictly necessary. I would have to test if things work regardless

Reply:
But the user could give a validation dataset with different units as the train one. In my opinion this should be not supported.

Reply:
Ok, then let's keep it this way. We're going to think more about these things once people start using the feature